### ISSUE TITLE
sintel的mask无法直接通过sam2获取

### ISSUE THREAD

**Issue by lovean1** [sunzunyi/M2FLOW#3]
作者你好，我尝试复现论文在Sintel上的表现，但是发现给出的py文件里无法直接使用，会出现无法使用png文件的错误，请问你是如何解决这一问题的，是否对sam2的代码进行了一定的修改？为了保证复现的性能，我希望能尽量还原sam2得到mask的过程。

<img width="1355" height="316" alt="Image" src="https://github.com/user-attachments/assets/b4f49ea3-1733-4437-a46c-8a24ea401200" />

**Comment by lovean1** [comment]
我对sam2的数据读取进行了一定的修改，使得代码可以运行，如果可以的话能否看一下这样的处理是否合理有效。
我对着这一行的代码进行了修改#104 inference_state = predictor_video.init_state(video_list=image_l)，直接将image_list作为输入，使得sam2不再判断png或jpg。

<img width="990" height="225" alt="Image" src="https://github.com/user-attachments/assets/49282fe6-4114-4821-b478-e3f924aa62c0" />

<img width="1143" height="701" alt="Image" src="https://github.com/user-attachments/assets/d21c130a-2463-4003-a692-cd49539294dd" />

**Comment by sunzunyi** [comment]
你好，你的做法没有问题，总之，使他能够处理png图片就行，下面是我的代码供参考，改自原始的load_video_frames_from_jpg_images

    def load_video_frames(
        video_path,
        image_size,
        offload_video_to_cpu,
        img_mean=(0.485, 0.456, 0.406),
        img_std=(0.229, 0.224, 0.225),
        async_loading_frames=False,
        compute_device=torch.device("cuda"),
    ):
        """
        Load the video frames from video_path. The frames are resized to image_size as in
        the model and are loaded to GPU if offload_video_to_cpu=False. This is used by the demo.
        """
        is_bytes = isinstance(video_path, bytes)
        is_str = isinstance(video_path, str)
        is_mp4_path = is_str and os.path.splitext(video_path)[-1] in [".mp4", ".MP4"]
        if is_bytes or is_mp4_path:
            return load_video_frames_from_video_file(
                video_path=video_path,
                image_size=image_size,
                offload_video_to_cpu=offload_video_to_cpu,
                img_mean=img_mean,
                img_std=img_std,
                compute_device=compute_device,
            )
        elif is_str and os.path.isdir(video_path):
            return load_video_frames_from_jpg_images(
                video_path=video_path,
                image_size=image_size,
                offload_video_to_cpu=offload_video_to_cpu,
                img_mean=img_mean,
                img_std=img_std,
                async_loading_frames=async_loading_frames,
                compute_device=compute_device,
            )
        elif isinstance(video_path, list):
            return load_video_frames_from_path_list(
                video_path=video_path,
                image_size=image_size,
                offload_video_to_cpu=offload_video_to_cpu,
                img_mean=img_mean,
                img_std=img_std,
                async_loading_frames=async_loading_frames,
                compute_device=compute_device,
            )
        else:
            raise NotImplementedError(
                "Only MP4 video and JPEG folder are supported at this moment"
            )

    def load_video_frames_from_path_list(
        video_path,
        image_size,
        offload_video_to_cpu,
        img_mean=(0.485, 0.456, 0.406),
        img_std=(0.229, 0.224, 0.225),
        async_loading_frames=False,
        compute_device=torch.device("cuda"),
    ):
        """
        Load the video frames from a directory of JPEG files ("<frame_index>.jpg" format).

        The frames are resized to image_size x image_size and are loaded to GPU if
        `offload_video_to_cpu` is `False` and to CPU if `offload_video_to_cpu` is `True`.

        You can load a frame asynchronously by setting `async_loading_frames` to `True`.
        """

        num_frames = len(video_path)
        img_paths = video_path
        img_mean = torch.tensor(img_mean, dtype=torch.float32)[:, None, None]
        img_std = torch.tensor(img_std, dtype=torch.float32)[:, None, None]

        if async_loading_frames:
            lazy_images = AsyncVideoFrameLoader(
                img_paths,
                image_size,
                offload_video_to_cpu,
                img_mean,
                img_std,
                compute_device,
            )
            return lazy_images, lazy_images.video_height, lazy_images.video_width

        images = torch.zeros(num_frames, 3, image_size, image_size, dtype=torch.float32)
        # for n, img_path in enumerate(tqdm(img_paths, desc="frame loading (JPEG)")):
        for n, img_path in enumerate(img_paths):
            images[n], video_height, video_width = _load_img_as_tensor(img_path, image_size)
        if not offload_video_to_cpu:
            images = images.to(compute_device)
            img_mean = img_mean.to(compute_device)
            img_std = img_std.to(compute_device)
        # normalize by mean and std
        images -= img_mean
        images /= img_std
        return images, video_height, video_width

**Comment by lovean1** [comment]
感谢回答，我已经在训练集上进行训练并得到了与论文相近的结果
